tests/test_inner.py

```diff
@@ -848,3 +867,59 @@ def test_workflow_plugin_results():
     assert workflow.postbuild_results == {'post_build_value': 'post_build_value_result'}
     assert workflow.prepub_results == {'pre_publish_value': 'pre_publish_value_result'}
     assert workflow.exit_results == {'exit_value': 'exit_value_result'}
+
+
+def test_cancel_build(request):
+    """
+    Verifies that exit plugins are executed when the build is cancelled
+    """
+    this_file = inspect.getfile(PreRaises)
+    mock_docker()
+    fake_builder = MockInsideBuilder()
+    flexmock(InsideBuilder).new_instances(fake_builder)
+
+    watch_pre = Watcher()
+    watch_prepub = Watcher()
+    watch_post = WatcherWithPause()
```

Comment:
Correct, I'll parametrize the test to make sure the signal is called during each phase